datafusion/core/src/physical_plan/projection.rs

```diff
@@ -410,12 +415,39 @@ fn stats_projection(
             .collect()
     });
 
-    Statistics {
-        is_exact: stats.is_exact,
-        num_rows: stats.num_rows,
-        column_statistics,
-        // TODO stats: knowing the type of the new columns we can guess the output size
-        total_byte_size: None,
+    let primitive_row_size = inner_exprs
+        .clone()
+        .into_iter()
+        .map(|e| match e.data_type(schema.as_ref()) {
+            Ok(data_type) => data_type.primitive_width(),
+            Err(_) => None,
+        })
+        .fold(Some(0usize), |init, v| match (init, v) {
+            (Some(l), Some(r)) => Some(l + r),
+            _ => None,
+        });
+
+    match (primitive_row_size, stats.num_rows) {
+        (Some(row_size), Some(row_count)) => {
+            Statistics {
+                is_exact: stats.is_exact,
+                num_rows: stats.num_rows,
+                column_statistics,
+                // Use the row_size * row_count as the total byte size
+                total_byte_size: Some(row_size * row_count),
+            }
+        }
+        _ => {
+            Statistics {
+                is_exact: stats.is_exact,
+                num_rows: stats.num_rows,
+                column_statistics,
+                // TODO stats: knowing the type of the new columns we can guess the output size
+                // If we can't get the exact statistics for the project
+                // Before we get the exact result, we just use the child status
+                total_byte_size: stats.total_byte_size,
```

Comment:
If type has changed `stats.is_exact` will not be true anymore (for `total_byte_size`).

Reply:
I think they are different things.

`stats.is_exact` is for the whole statistic. Now we should need a status to present just `column_statistics`.

Reply:
Hm I think you're right.
The `is_exact` is currently a bit confusing though, I think it would be nice if we have a exact/inexact specifier for both row/byte statistics.